From 0e1699ce00334c597571b5894f378d7745fdb1e7 Mon Sep 17 00:00:00 2001
From: Daniel Veillard <veillard@redhat.com>
Date: Mon, 9 Nov 2015 18:16:00 +0800
Subject: [PATCH] xmlStopParser reset errNo
To: libvir-list@redhat.com

I had used it in contexts where that information ought to be preserved

Signed-off-by: Daniel Veillard <veillard@redhat.com>
---
 parser.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/parser.c b/parser.c
index a817850..9555445 100644
--- a/parser.c
+++ b/parser.c
@@ -6577,6 +6577,7 @@ xmlParseConditionalSections(xmlParserCtxtPtr ctxt) {
 	if (RAW != '[') {
 	    xmlFatalErr(ctxt, XML_ERR_CONDSEC_INVALID, NULL);
 	    xmlStopParser(ctxt);
+	    ctxt->errNo = XML_ERR_CONDSEC_INVALID;
 	    return;
 	} else {
 	    if (ctxt->input->id != id) {
@@ -6639,6 +6640,7 @@ xmlParseConditionalSections(xmlParserCtxtPtr ctxt) {
 	if (RAW != '[') {
 	    xmlFatalErr(ctxt, XML_ERR_CONDSEC_INVALID, NULL);
 	    xmlStopParser(ctxt);
+	    ctxt->errNo = XML_ERR_CONDSEC_INVALID;
 	    return;
 	} else {
 	    if (ctxt->input->id != id) {
@@ -6696,6 +6698,7 @@ xmlParseConditionalSections(xmlParserCtxtPtr ctxt) {
     } else {
 	xmlFatalErr(ctxt, XML_ERR_CONDSEC_INVALID_KEYWORD, NULL);
 	xmlStopParser(ctxt);
+	ctxt->errNo = XML_ERR_CONDSEC_INVALID_KEYWORD;
 	return;
     }
 
-- 
2.5.0

